crates/autopilot/src/domain/settlement/trade/math.rs

```diff
@@ -91,78 +90,63 @@ impl Trade {
                 bought.checked_sub(limit_buy).ok_or(error::Math::Negative)
             }
         }
-        .map(|surplus| eth::Asset {
-            token: self.surplus_token(),
-            amount: surplus.into(),
-        })
+        .map(eth::TokenAmount)
     }
 
     /// Surplus based on custom clearing prices returns the surplus after all
     /// fees have been applied.
     pub fn surplus_in_ether(&self, prices: &auction::Prices) -> Result<eth::Ether, Error> {
         let surplus = self.surplus_over_limit_price()?;
         let price = prices
-            .get(&surplus.token)
-            .ok_or(Error::MissingPrice(surplus.token))?;
+            .get(&self.surplus_token())
+            .ok_or(Error::MissingPrice(self.surplus_token()))?;
 
-        Ok(price.in_eth(surplus.amount))
+        Ok(price.in_eth(surplus))
     }
 
-    /// Total fee (protocol fee + network fee). Equal to a surplus difference
```

Comment:
Silly question: what is the difference between network and gas fees in this context?

Reply:
It's a synonym. I don't think we ever decided officially how to call this fee, because of the following:

Note how the total fee (that we now call `executed_fee`) is calculated - as a difference between UCP and custom prices of an order. 

And let's assume that protocol fees are not taken = 0.

Normally, if a solver calculates that for an order it needs to spend 10$ for gas costs, solver will report such custom prices and UCP prices so that their difference represents these 10$ and are stored in the settlement contract.

But, solvers don't have to do it. They can report 20$ which effectively means their custom prices are worse and their solution is less competitive. This is not a reasonable to do, because solvers don't get anything out of it, but just for the sake of example.

Another more realistic example is that solvers report 5$ (so less than expected and let's assume realized gas costs) which means their custom prices are better and their solution is more competitive. Solvers are incentivized to report 5$ because if they win, they will be rewarded with (let' say) 50$ of COW. They are even incentivized to report 0$ in this case and they are still positive.

And even more, solvers are not forced by any mechanism to even report UCP prices anymore, once colocated. They can always report UCP = custom prices and we wouldn't be able to estimate how much was spent for gas costs. In this case, solvers can bake in the gas costs in custom prices, offering a bit smaller custom prices then expected, and we again can't deduct the gas costs.

So, should we call this fee `gas fee`, or `network fee`, or `solver fee`?  I don't know. IMO, this fee is a relic from non-colocated world and we should seriously think about deprecating it completely. The reason we didn't do it yet is because it's a non-trivial task to deduct approximate gas costs per order in a batched settlement, and we need it for frontend to show it to the users.

Reply:
Actually, thanks for bringing this up. After some thinking, I don't think my previous implementation was good enough. It definitely changes the expectation on the final `executed_fee` value in all of these special cases, so I reverted back to previous implementation - `executed_fee` and `protocol_fee` being completely decoupled.

Bottom line, assuming that `executed_fee` represents protocol_fee + gas fee and always >= than `protocol_fee` is just not guaranteed.

Reply:
Oh, thanks a lot. That is much clearer now 🙌